packages/backend/src/apps/paysg/actions/create-payment-form-submission/schema.ts

```diff
@@ -0,0 +1,87 @@
+import emailValidator from 'email-validator'
+import { z } from 'zod'
+
+import { FORM_ID_LENGTH } from '@/apps/formsg/common/constants'
+
+import { normalizeSpecialChars } from '../create-payment/normalize-special-chars'
+
+const PAYMENT_FORM_LINK_REGEX = /^https:\/\/(staging.)?pay.gov.sg\/forms\/(.*)$/
+
+export const requestSchema = z.object({
+  formId: z
+    .string()
+    .trim()
+    .min(1, { message: 'Specify a form link' })
+    .transform((value, context) => {
+      const match = value.match(PAYMENT_FORM_LINK_REGEX)
+
+      if (!match || !match[0]) {
+        context.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: 'Invalid payment form link',
+        })
+        return z.NEVER
+      }
+
+      return match[2]
+    }),
+  formsg_form_id: z
+    .string()
+    .trim()
+    .length(FORM_ID_LENGTH, { message: 'Specify a valid FormSG form ID' }),
+  formsg_submission_id: z
+    .string()
+    .trim()
+    .min(1, { message: 'Specify a submission ID' }),
+  nonce: z.string().trim().min(1, { message: 'Specify a nonce field' }),
+  payer_name: z
+    .string()
+    .trim()
+    .min(1, { message: 'Empty payer name' })
+    .max(255, { message: 'Payer name cannot be more than 255 characters' })
+    .transform((value) => normalizeSpecialChars(value)),
+  payer_email: z
+    .string()
+    .trim()
+    .max(255, { message: 'Payer email cannot be more than 255 characters' })
+    .transform((value, context) => {
+      if (!value || value.length === 0) {
+        return undefined
+      }
+      if (!emailValidator.validate(value)) {
+        context.addIssue({
+          code: z.ZodIssueCode.custom,
+          message: 'Invalid payer email',
+        })
+        return z.NEVER
+      }
+
+      return value
+    })
+    .optional(),
+  amount_in_cents: z
+    .string()
+    .trim()
+    .min(1, { message: 'Empty payment amount' })
+    .pipe(
+      z.coerce
+        .number()
+        .int('Payment amount must be round number')
```

Comment:
```suggestion
        .number({
          invalid_type_error: 'Payment amount must be a number',
        })
```
suggest to add this message, otherwise error message shows "Expected number, received NaN"

Reply:
done in [9cf3cac](https://github.com/opengovsg/plumber/pull/870/commits/9cf3cacb42b3e9221d11c6e307b27b7a748042b5)